websites/F/Floatplane/presence.ts

```diff
@@ -0,0 +1,221 @@
+const presence = new Presence({
+    clientId: "889918462477095012"
+  }),
+  browsingTimestamp = new Date().getTime();
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "logo",
+      startTimestamp: browsingTimestamp
+    },
+    pathnames = location.pathname.split("/");
+
+  if (pathnames[1] === "") {
+    //Uploads
+    const subCount =
+        document.querySelector(".simplebar-content")?.children.length,
+      showCount = await presence.getSetting("subscriptions");
+
+    presenceData.details = "Vieweing uploads";
+
+    if (showCount && subCount > 0) {
+      presenceData.state = `${subCount - 1} ${
+        subCount > 2 ? "Subscriptions" : "Subscription"
+      }`;
+    }
+  } else if (pathnames[1] === "discover") {
+    //Homepage
+
+    presenceData.details = "Viewing Homepage";
+    presenceData.state = "Say Hello To Floatplane";
+  } else if (pathnames[1] === "browse") {
+    //Browsing
+    const searchTerm = (
+        document.querySelector(".search-bar") as HTMLInputElement
+      )?.value,
+      channelCount = document.querySelector(".creator-cards")?.children.length;
+
+    presenceData.details = "Browsing";
+    presenceData.state = `${channelCount} Channels`;
+    presenceData.smallImageKey = "reading";
+    presenceData.smallImageText = "Browsing";
+
+    if (searchTerm) {
+      let stringFormated;
+
+      switch (channelCount) {
+        case 1:
+          if (document.querySelector(".search-not-found")) {
+            stringFormated = "No Results";
+            break;
+          }
+          stringFormated = "1 Channel";
+          break;
+
+        default:
+          stringFormated = `${channelCount} Channels`;
+      }
+
+      presenceData.details = `Searching: ${searchTerm}`;
+      presenceData.state = stringFormated;
+      presenceData.smallImageKey = "search";
+      presenceData.smallImageText = "Searching";
+    }
+  } else if (pathnames[1] === "channel") {
+    //Viewing a channel
+    const channelTitle = document.querySelector(".channel-title")?.textContent;
+
+    //Wait for page to load
+    if (!channelTitle) return;
+
+    presenceData.details = "Viewing channel:";
+    presenceData.state = channelTitle;
+
+    presenceData.largeImageKey = pathnames[2].toLowerCase();
+
+    presenceData.smallImageKey = "logo";
+    presenceData.smallImageText = document.title;
+
+    presenceData.buttons = [{ label: "View Channel", url: location.href }];
+
+    if (pathnames[3] === "live") {
+      //Stream
+      const title = document.querySelector(".title-text")?.textContent,
+        video = document.querySelector("video") as HTMLVideoElement;
+
+      if (!title || !video) return;
+
+      presenceData.details = title;
+
+      delete presenceData.startTimestamp;
+
```

Comment:
Makes more sense to do it this way as you're assigning values on the line below
```suggestion
      delete presenceData.startTimestamp;
      presenceData.details = title;
```